Kernel/Coredump.cpp

```diff
@@ -240,8 +242,12 @@ ErrorOr<void> Coredump::write_regions()
                 // If the current page is not backed by a physical page, we zero it in the coredump file.
                 return UserOrKernelBuffer::for_kernel_buffer(zero_buffer);
             }();
-            TRY(m_description->write(src_buffer.value(), PAGE_SIZE));
+            TRY(src_buffer.value().read(buffer->bytes().slice(i * PAGE_SIZE, PAGE_SIZE)));
         }
+
+        lock.unlock();
```

Comment:
Since you don't need anything from the `real_region` to allocate the buffer, consider moving the buffer creation the spinlock is acquired, then just use a normal scope to enclose the spinlock.